src/plugin.ts

```diff
@@ -9,49 +9,45 @@ import { adjustCanvas, parseImage } from './utils/image'
 import { logger } from './utils/logger'
 import { type DiffOption } from './command'
 
-export type UpdateSnapshotOptions = {
-  screenshotName: string
-  specName: string
-  screenshotAbsolutePath: string
-  baseDirectory?: string
-}
-
-export type CompareSnapshotsOptions = {
```

Comment:
Actually, this is needed because on `updateSnapshot` we are receiving this shape and you are now saying that you accept `VisualRegressionOptions` which is a broader type.

You can set the type by using helper Pick, like:

```ts
export type UpdateSnapshotOptions = Pick<VisualRegressionOptions, 'screenshotName' | 'specName' | 'screenshotAbsolutePath' | 'baseDirectory'>

```

Reply:
But why would narrowing down types be a better option?

p.s. added the `Pick` in the latest commit

Reply:
Nice 👌

> But why would narrowing down types be a better option?

Because if you look at the internals of updateSnapshot, you will see that you are only using a subset of the keys, that is why you want to state to use only needed keys
If you don't do so then you can later end up passing keys that you are never going to use it and will be confusing and hard to read and understand